File: AgentieTurism/AgentieTurism/repository/TripRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AgentieTurism.domain;

namespace AgentieTurism.repository
{
    interface TripRepository:Repository<long,Trip>
    {
        IEnumerable<Trip> findTripsByDate(DateTime minDate, DateTime maxDate);
        IEnumerable<Trip> findTripsByName(String name);
        IEnumerable<Trip> findTripsByNameAndHours(String name, int minTime, int maxTime);
    }
}

File: TourismAgency/AgencyPersistance/src/main/java/repository/TripRepository.java
package repository;

import domain.Trip;

import java.time.LocalDate;
import java.time.LocalTime;

public interface TripRepository extends Repository<Long, Trip>{
    Iterable<Trip> findTripsByDate(LocalDate minDate, LocalDate maxDate);
    Iterable<Trip> findTripsByName(String name);
    Iterable<Trip> findTripsByNameAndHours(String name , LocalTime minTime,LocalTime maxTime);
}
